Arduino.h: error on max macro definition

### Hardware:
Board:							ESP32 Dev Module
Core Installation/update date:			up to date as of 06/08/2018
IDE name:							VSCode
Flash Frequency:					80Mhz
Upload Speed:						921600


### Description:
I am working with a TTGO esp32 module and I'm using vectors in my code, and it compiled fine up to a certain point. Then, out of the blue, it suddenly started giving me this error when trying to upload the code to the board: 

```
c:\users\ivano\documents\arduino\hardware\espressif\esp32\tools\xtensa-esp32-elf\xtensa-esp32-elf\include\c++\5.2.0\bits\stl_vector.h: 
In member function 'std::vector<_Tp, _Alloc>::size_type std::vector<_Tp, _Alloc>::_M_check_len(std::vector<_Tp, _Alloc>::size_type, const char*) const':

C:\Users\Ivano\Documents\Arduino\hardware\espressif\esp32\cores\esp32/Arduino.h:182:18: error: expected unqualified-id before '(' token

 #define max(a,b) ((a)>(b)?(a):(b))
```

To get it to compile again I had to undefine min and max after including Arduino.h

```c++
#include <Arduino.h>
#undef min
#undef max
```

It might be worth noting that I'm also including `vector` in another file but commenting that include out did not change the problem. 

Is there a reason why this should happen?

--- a/cores/esp32/Arduino.h
+++ b/cores/esp32/Arduino.h
@@ -75,7 +75,6 @@
 
 #define abs(x) ((x)>0?(x):-(x))
 #define constrain(amt,low,high) ((amt)<(low)?(low):((amt)>(high)?(high):(amt)))
-#define round(x)     ((x)>=0?(long)((x)+0.5):(long)((x)-0.5))
 #define radians(deg) ((deg)*DEG_TO_RAD)
 #define degrees(rad) ((rad)*RAD_TO_DEG)
 #define sq(x) ((x)*(x))
@@ -146,6 +145,9 @@ void shiftOut(uint8_t dataPin, uint8_t clockPin, uint8_t bitOrder, uint8_t val);
 #ifdef __cplusplus
 }
 
+#include <algorithm>
+#include <cmath>
+
 #include "WCharacter.h"
 #include "WString.h"
 #include "Stream.h"
@@ -157,6 +159,12 @@ void shiftOut(uint8_t dataPin, uint8_t clockPin, uint8_t bitOrder, uint8_t val);
 #include "Udp.h"
 #include "HardwareSerial.h"
 #include "Esp.h"
+
+using std::isinf;
+using std::isnan;
+using std::max;
+using std::min;
+using ::round;
 
 uint16_t makeWord(uint16_t w);
 uint16_t makeWord(byte h, byte l);
@@ -175,12 +183,6 @@ extern "C" void configTzTime(const char* tz,
 // WMath prototypes
 long random(long);
 #endif /* __cplusplus */
-
-#ifndef _GLIBCXX_VECTOR
-// arduino is not compatible with std::vector
-#define min(a,b) ((a)<(b)?(a):(b))
-#define max(a,b) ((a)>(b)?(a):(b))
-#endif
 
 #define _min(a,b) ((a)<(b)?(a):(b))
 #define _max(a,b) ((a)>(b)?(a):(b))

--- a/libraries/WiFi/src/WiFiClient.h
+++ b/libraries/WiFi/src/WiFiClient.h
@@ -23,8 +23,6 @@
 
 #include "Arduino.h"
 #include "Client.h"
-#undef min
-#undef max
 #include <memory>
 
 class WiFiClientSocketHandle;

--- a/libraries/WiFi/src/WiFiGeneric.cpp
+++ b/libraries/WiFi/src/WiFiGeneric.cpp
@@ -46,9 +46,6 @@ extern "C" {
 } //extern "C"
 
 #include "esp32-hal-log.h"
-
-#undef min
-#undef max
 #include <vector>
 
 #include "sdkconfig.h"
@@ -156,8 +153,8 @@ static bool espWiFiStart(bool persistent){
         return false;
     }
     _esp_wifi_started = true;
-    system_event_t event; 
-    event.event_id = SYSTEM_EVENT_WIFI_READY; 
+    system_event_t event;
+    event.event_id = SYSTEM_EVENT_WIFI_READY;
     WiFiGenericClass::_eventCallback(nullptr, &event);
 
     return true;
@@ -496,7 +493,7 @@ bool WiFiGenericClass::mode(wifi_mode_t m)
     } else if(cm && !m){
         return espWiFiStop();
     }
-    
+
     esp_err_t err;
     err = esp_wifi_set_mode(m);
     if(err){

--- a/libraries/WiFi/src/WiFiMulti.h
+++ b/libraries/WiFi/src/WiFiMulti.h
@@ -27,8 +27,6 @@
 #define WIFICLIENTMULTI_H_
 
 #include "WiFi.h"
-#undef min
-#undef max
 #include <vector>
 
 typedef struct {

